### PR TITLE
Add new unittests for Quickstart docs

### PR DESCRIPTION
**Pull request by young-k**
**Patch description**

Add test for quickstart, to ensure that any users following it will not run into any errors.

**Testing steps**

Added a new CircleCI test.

**Logs**

N/A

**Other information**

There should be a way to keep these in sync in the future, but that is most likely separate from this PR.

**Data tests (if applicable)**

N/A

### CODE REVIEW

**Comment by stephenroller** [comment]
Also, please update the PR message.

File: requirements.txt [hunk 1]
```diff
@@ -22,6 +23,7 @@ sh
 sphinx
 sphinx_rtd_theme
 tqdm
+unidecode
```
[the remark this answers is not in the document]
Comment (young-k) [reply requirements.txt]: This and the above were required for the twitter dataset, but the dataset was leading to an OOM on CircleCI, and @klshuster suggested using `convai2` instead, so I'll remove these changes!

[the remark this answers is not in the document]
Comment (young-k) [reply requirements.txt]: It's necessary for the `twitter` dataset and currently not in the quickstart docs –– I can add it to the docs as a step prior to running the dataset!

File: tests/test_quickstart.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# view a task & train a model
```
Comment (stephenroller) [inline tests/test_quickstart.sh]: add this to the beginning:

```bash
#!/bin/bash
# Copyright (c) Facebook, Inc. and its affiliates.
# This source code is licensed under the MIT license found in the
# LICENSE file in the root directory of this source tree.

set -e  # error and exit on any failure
```

File: tests/test_quickstart.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+pip3 install emoji unidecode
+python examples/display_data.py -t twitter
+python examples/train_model.py -t twitter -mf /tmp/tr_twitter -m transformer/ranker -bs 10 -vtim 3600 -cands batch -ecands batch --data-parallel True --num-epochs 0.01
+python examples/eval_model.py -t twitter -m transformer/ranker -mf models:convai2/seq2seq/convai2_self_seq2seq_model
+python examples/display_model.py -t twitter -mf /tmp/tr_twitter -ecands batch
+
+# add a simple model
+mkdir parlai/agents/parrot
+touch parlai/agents/parrot/parrot.py
+echo "from parlai.core.torch_agent import TorchAgent, Output" >> parlai/agents/parrot/parrot.py
```
Comment (stephenroller) [inline tests/test_quickstart.sh]: bash supports multiline strings:

```bash
cat > parlai/agents/parrot/parrot.py <<- EOF
from parlai.core.torch_agent import TorchAgent, Output
class ParrotAgent(TorchAgent):
    ....
EOF
```

File: tests/test_quickstart.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+echo "class ParrotAgent(TorchAgent):" >> parlai/agents/parrot/parrot.py
+echo "    def train_step(self, batch):" >> parlai/agents/parrot/parrot.py
+echo "        pass" >> parlai/agents/parrot/parrot.py
+echo "    def eval_step(self, batch):" >> parlai/agents/parrot/parrot.py
+echo "        return Output([self.dict.vec2txt(row) for row in batch.text_vec])" >> parlai/agents/parrot/parrot.py
+python examples/display_model.py -t babi:task10k:1 -m parrot
```
Comment (stephenroller) [inline tests/test_quickstart.sh]: I wonder if, for each of these commands, we can also grep for the command in the quickstart.rst, to detect if it goes out of date.

File: .circleci/config.yml [hunk 5]
```diff
@@ -182,6 +182,9 @@ jobs:
       - run:
           name: Unit tests (GPU; pytorch 1.1)
           command: python setup.py test -s tests.suites.unittests -v
+      - run:
+          name: Quickstart unit tests (GPU; pytorch 1.1)
```
Comment (stephenroller) [inline .circleci/config.yml]: perfect

File: docs/source/tutorial_quick.rst [hunk 6]
```diff
@@ -79,7 +79,14 @@ including a full list of `tasks <http://parl.ai/docs/tasks.html>`_ and
 `instructions <http://parl.ai/docs/tutorial_basic.html#training-and-evaluating-existing-agents>`_
 on specifying arguments for training and evaluation (like the ``-t <task>`` argument used here).
 
-Let's begin again by printing the first few examples.
+First, we need to install some dependencies to build the twitter dataset.
+
+.. code-block:: bash
+
+  # install necessary dependencies
+  pip3 install emoji unidecode
```
Comment (stephenroller) [inline docs/source/tutorial_quick.rst]: I'm still confused. If they need to be in requirements.txt, let's put them there. I'm just surprised. This tutorial shouldn't have any extra installation steps though.

File: docs/source/tutorial_quick.rst [hunk 7]
```diff
@@ -102,13 +109,13 @@ The Twitter task is quite large, and validation is run by default after each epo
...
-perhaps to compare the results of our newly trained Transformer against a pretrained seq2seq baseline from our `Model Zoo <http://parl.ai/docs/zoo.html>`_,
+perhaps to compare the results of our newly trained Transformer against a pretrained ``convai2`` seq2seq baseline from our `Model Zoo <http://parl.ai/docs/zoo.html>`_,
 we could do the following:
 
 .. code-block:: bash
 
-  # Evaluate seq2seq model trained on twitter from our model zoo
-  python examples/eval_model.py -t twitter -m legacy:seq2seq:0 -mf models:twitter/seq2seq/twitter_seq2seq_model
+  # Evaluate seq2seq model trained on convai2 from our model zoo
+  python examples/eval_model.py -t twitter -m legacy:seq2seq:0 -mf models:convai2/seq2seq/convai2_self_seq2seq_model
```
Comment (stephenroller) [inline docs/source/tutorial_quick.rst]: great, thank you.

File: tests/test_quickstart.sh (hunk 4 above)
Comment (stephenroller) [reply tests/test_quickstart.sh]: (That's for the next PR)

File: docs/source/tutorial_quick.rst (hunk 6 above)
Comment (stephenroller) [reply docs/source/tutorial_quick.rst]: Let's just add it to requirements.txt and move on.

**Pull request merged**